src/ipolatev.F90

```diff
@@ -702,5 +703,130 @@ subroutine ipolatev_grib1_single_field(ip,ipopt,kgdsi,kgdso,mi,mo,km,ibi,li,ui,v
+  !> gridmod data structure:
+  !> - 00 - Equidistant Cylindrical
+  !> - 01 - Rotated Equidistant cylindrical. "e" and non-"e" staggered
+  !> - 10 - Mercator Cyclindrical
+  !> - 20 - Polar Stereographic azimuthal
+  !> - 30 - Lambert Conformal Conical
+  !> - 40 - Gaussian Equidistant Cyclindrical
+  !>
+  !> @param[in] igdtmpli Grid definition template array input grid.
+  !> Corresponds to the gfld%igdtmpl component of the NCEPLIBS-g2
+  !> gridmod data structure
+  !>
+  !> @param[in] igdtleni Number of elements of the grid definition
+  !> template array for the input grid. Corresponds to the gfld%igdtlen
+  !> component of the ncep g2 library gridmod data structure.
+  !>
+  !> @param[in] igdtnumo Grid definition template number for the output grid.
+  !> Corresponds to the gfld%igdtnum component of the
+  !> ncep g2 library gridmod data structure.
+  !> See "igdtnumi" for specific template definitions.
+  !> Note: igdtnumo<0 means interpolate to random station points.
+  !>
+  !> @param[in] igdtmplo Grid definition template array for the output grid.
+  !> Corresponds to the gfld%igdtmpl component of the ncep g2 library
+  !> gridmod data structure.
+  !> See "igdtmpli" for definition of array elements.
+  !>
+  !> @param[in] igdtleno Number of elements of the grid definition
+  !> template array for the output grid. Corresponds to the gfld%igdtlen
+  !> component of the ncep g2 library gridmod data structure.
+  !>
+  !> @param[in] mi    Skip number between input grid fields if km>1 or
+  !> dimension of input grid fields if km=1.
+  !> @param[in] mo    Skip number between output grid fields if km>1
+  !> or dimension of output grid fields if km=1.
+  !> @param[in] km    Number of fields to interpolate.
+  !> @param[in] ibi   Input bitmap flags.
+  !> @param[in] li    Input bitmaps (if respective ibi(k)=1).
+  !> @param[in] ui    Input u-component fields to interpolate.
+  !> @param[in] vi    Input v-component fields to interpolate.
+  !> @param[out] no Number of output points (only if kgdso(1)<0).
+  !> @param[inout] rlat Output latitudes in degrees (if kgdso(1)<0).
+  !> @param[inout] rlon Output longitudes in degrees (if kgdso(1)<0).
+  !> @param[inout] crot Vector rotation cosines (if igdtnumo>=0).
+  !> @param[inout] srot Vector rotation sines (if igdtnumo>=0).
+  !> @param[out] ibo Output bitmap flags.
+  !> @param[out] lo  Output bitmaps (always output).
+  !> @param[out] uo  Output u-component fields interpolated.
+  !> @param[out] vo  Output v-component fields interpolated.
+  !> @param[out] iret Return code.
+  !> - 0 Successful interpolation.
+  !> - 1 Unrecognized interpolation method.
+  !> - 2 Unrecognized input grid or no grid overlap.
+  !> - 3 Unrecognized output grid.
+  !> - 1x Invalid bicubic method parameters.
+  !> - 3x Invalid budget method parameters.
+  !> - 4x Invalid spectral method parameters.
+  !>
+  !> @author Kyle Gerheiser @date July 2021
```

Comment:
Did Kyle write this or did you? Who is the author? Since you are adding the function, I believe it should be you, unless this is some code that Kyle wrote before he left and never checked in...